tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+
+''' % (soid, soid, soname, soid, soid, snname, t_fname_out, baseline_dir, baseline_dir)
+                            if soname != snname:
+                                shutil.move(t_fname_out, t_fname_out_fail)
+                            self.assertEqual(soname, snname, msg=fail_msg)
+            feature_names_match = True
+        else:
+            feature_names_match = False
+            shutil.move(t_fname_out, t_fname_out_fail)
+
+        self.assertEqual(feature_names_match, True)
+        return True
+
+    def test_tsfresh_baseline_json(self):
+        """
+        This test compares the calculated feature names AND values in the
+        transposed csv to the feature names AND values in the baseline
+        transposed csv.  It outputs the differences if the test fails.
+        """
+        start = self.test_tsfresh_baseline()
+        self.assertEqual(start, True)
+
+        if os.path.isfile(t_fname_out):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        df = pd.read_csv(
+            t_fname_out, delimiter=',', header=0,
+            names=['feature_name', 'value'])
+        baseline_df = pd.read_csv(
+            baseline_ts_json_baseline, delimiter=',', header=0,
+            names=['feature_name', 'value'])
+
+        # There is a more friendly user output than DataFrames comparison
+        # dataframes_equal = df.equals(baseline_df)
+        dataframes_equal = True
+        if not dataframes_equal:
+            df1 = df
+            df2 = baseline_df
+            ne = (df1 != df2).any(1)
+            ne_stacked = (df1 != df2).stack()
+            changed = ne_stacked[ne_stacked]
+            changed.index.names = ['id', 'col']
+            difference_locations = np.where(df1 != df2)
+            changed_from = df1.values[difference_locations]
+            changed_to = df2.values[difference_locations]
+            _fail_msg = pd.DataFrame(
+                {'from': changed_from, 'to': changed_to}, index=changed.index)
+            fail_msg = 'Baseline comparison failed - %s' % _fail_msg
+            self.assertEqual(dataframes_equal, True, msg=fail_msg)
+        # self.assertEqual(dataframes_equal, True)
+
+        # This is more friendly user output than the above DataFrames comparison
+        calculated_features = []
+        sortedlist_calcf = []
+        with open(t_fname_out, 'rt') as fr:
+            reader = csv.reader(fr, delimiter=',')
+            sortedlist_calcf = sorted(reader, key=lambda row: row[0], reverse=True)
+
+        for i, line in enumerate(sortedlist_calcf):
+            calculated_features.append([str(line[0]), str(line[1])])
+
+        baseline_features = []
+        sortedlist_baseline = []
+        with open(baseline_ts_json_baseline, 'rt') as fr:
+            reader = csv.reader(fr, delimiter=',')
+            sortedlist_baseline = sorted(reader, key=lambda row: row[0], reverse=True)
+
```

Comment:
Actually I liked you way of using the pandas dataframe more instead of going through lines in the CSV (which is more low-level).
You can still iterate through the lines in the pandas dataframe maybe.